lib/apiaria.rb

```diff
@@ -28,4 +28,27 @@ def get_mentor_availability(mentor_id)
     JSON.parse(response.body)
   end
   
+  def get_messages(pageNum = nil)
+    unless pageNum
+      response = self.class.get("/message_threads",  headers: { "authorization" => @auth_token })
+      JSON.parse(response.body)
+    else
+      url = "/message_threads?page=" + pageNum.to_s
+      response = self.class.get(url,  headers: { "authorization" => @auth_token })
+      JSON.parse(response.body)
+    end
+  end
+
+  def create_message(sender, recipientId, body, options = {})
```

Comment:
Hi @praveenyr the options hash is a good way to implement the optional parameters but that is outside of the scope of the checkpoint. Just because Bloc has attributes it can take on an endpoint, does mean you need to solve for them. In this case, we just need to 3 required field + text to make it useful. 

I'm consistently receiving messages from you on my side so what you changed, it is working :)